addons/mail/static/src/models/messaging_notification_handler.js

```diff
@@ -446,12 +460,21 @@ registerModel({
          * @param {integer[]} param0.messag_ids
          */
         _handleNotificationMessageDelete({ message_ids }) {
+            let decrementAmounts = new Array(this.mailboxInfos.length).fill(0);
             for (const id of message_ids) {
                 const message = this.messaging.models['Message'].findFromIdentifyingData({ id });
                 if (message) {
+                    this.mailboxInfos.forEach((mailboxInfo, idx) => {
+                        if (mailboxInfo.hasMessage(message)) {
+                            decrementAmounts[idx]++;
+                        }
+                    });
                     message.delete();
                 }
             }
+            this.mailboxInfos.forEach((mailBoxInfo, idx) => {
+                mailBoxInfo.mailbox.update({ counter: decrement(decrementAmounts[idx]) });
+            });
```

Comment:
I think just hardcode one case for inbox and one for starred, it will be easier to read than indirection through these loops and arrays.

Reply:
Yes it is simpler, especially if we only have 2 counters to update.